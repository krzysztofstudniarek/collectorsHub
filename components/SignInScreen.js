import React from 'react';
import {
  AsyncStorage,
  Button,
  StyleSheet,
  View
} from 'react-native';
import DropdownAlert from 'react-native-dropdownalert';
import t from 'tcomb-form-native';
const Form = t.form.Form;

import Amplify, { Auth } from 'aws-amplify';
import aws_exports from './../aws-exports';

Amplify.configure(aws_exports);

const User = t.struct({
    username: t.String,
    password: t.String,
  });

const Options = {
    auto: 'placeholders',
    fields: {
      password: {
        password: true,
        secureTextEntry: true
      }
    }
  };

export default class SignInScreen extends React.Component {
    static navigationOptions = {
      title: 'Please sign in',
    };

    constructor(props) {
        super(props)
    }

    render() {
        return (
            <View>
                <Form 
                ref={c => this._form = c} 
                type={User}
                options={Options} />
                <Button title="Sign in!" onPress={this._signInAsync} />
                <DropdownAlert ref={ref => this.dropdown = ref} />
            </View>
        );
    }
  
    _signInAsync = async () => {
        const value = this._form.getValue();
        Auth.signIn(value.username, value.password)
            .then(user => {
                AsyncStorage.setItem('userToken', user);
                this.props.navigation.navigate('App');
            })
            .catch(error => {
                this.dropdown.alertWithType('error', 'Error', error.message);
            });

    };
  }