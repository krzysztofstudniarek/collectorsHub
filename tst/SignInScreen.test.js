import React from 'react';
import SignInScreen from '../src/components/SignInScreen';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<SignInScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});