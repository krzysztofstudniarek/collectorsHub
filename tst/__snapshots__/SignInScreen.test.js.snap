// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel={null}
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Username"
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 10,
        }
      }
    >
      <View
        style={Object {}}
      >
        <TextInput
          accessibilityLabel={null}
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#cccccc",
              "borderRadius": 4,
              "borderWidth": 1,
              "color": "#000000",
              "fontSize": 17,
              "height": 36,
              "marginBottom": 5,
              "paddingHorizontal": 7,
              "paddingVertical": 7,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Sign in!
      </Text>
    </View>
  </View>
</View>
`;
